integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -0,0 +1,1361 @@
+          const subs = this.realm.subscriptions;
+          expect(subs.empty).to.be.true;
+
+          const sub = subs.update((mutableSubs) => {
+            return mutableSubs.add(this.realm.objects(FlexiblePersonSchema.name));
+          });
+
+          expect(subs.empty).to.be.false;
+
+          subs.update((mutableSubs) => {
+            mutableSubs.removeSubscription(sub);
+          });
+
+          expect(subs.empty).to.be.true;
+        });
+      });
+
+      describe("array-like access", function () {
+        it("returns an empty array if there are no subscriptions", function (this: RealmContext) {
+          const subs = this.realm.subscriptions;
+          expect(subs).to.have.length(0);
+        });
+
+        it("returns an array of Subscription objects", async function (this: RealmContext) {
+          addSubscriptionForPerson(this.realm);
+          const { subs } = addSubscription(
+            this.realm,
+            this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10"),
+          );
+
+          expect(subs).to.have.length(2);
+          expect(subs.every((s) => s instanceof Realm.App.Sync.Subscription)).to.be.true;
+        });
+
+        it("is an immutable snapshot of the subscriptions when snapshot was called", function (this: RealmContext) {
+          const { subs } = addSubscriptionForPerson(this.realm);
+          const snapshot = subs;
+          expect(snapshot).to.have.length(1);
+
+          addSubscription(this.realm, this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10"));
+
+          expect(snapshot).to.have.length(1);
+        });
+      });
+
+      describe("#findByName", function () {
+        it("returns null if the named subscription does not exist", function (this: RealmContext) {
+          expect(this.realm.subscriptions.findByName("test")).to.be.null;
+        });
+
+        it("returns the named subscription", function (this: RealmContext) {
+          const { subs, sub } = addSubscriptionForPerson(this.realm, { name: "test" });
+
+          expect(subs.findByName("test")).to.deep.equal(sub);
+        });
+      });
+
+      describe("#find", function () {
+        it("returns null if the query is not subscribed to", function (this: RealmContext) {
+          expect(this.realm.subscriptions.findByQuery(this.realm.objects(FlexiblePersonSchema.name))).to.be.null;
+        });
+
+        it("returns a query's subscription by reference", function (this: RealmContext) {
+          const { subs, sub, query } = addSubscriptionForPerson(this.realm);
+
+          expect(subs.findByQuery(query)).to.deep.equal(sub);
+        });
+
+        it("returns a filtered query's subscription", function (this: RealmContext) {
+          const query = this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10");
+          const { subs, sub } = addSubscription(this.realm, query);
+
+          expect(subs.findByQuery(query)).to.deep.equal(sub);
```

Comment:
We are adding a subscription for this filtered query (`this.realm.objects(FlexiblePersonSchema.name).filtered("age > 10")` – we call `addSubscription(.., query)`) rather than the generic person query